lib/rubocop/cop/sorbet/scope_helper_position.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Sorbet
+      # This cop ensures that helpers such as abstract!, interface! and final!
+      # are invoke before any method definitions
+      #
+      # @example
+      #
+      #   # bad
+      #   module Interface
+      #     extend T::Sig
+      #     extend T::Helpers
+      #
+      #     sig { returns(String) }
+      #     def foo; end
+      #
+      #     interface!
+      #   end
+      #
+      #   # good
+      #   module Interface
+      #     extend T::Sig
+      #     extend T::Helpers
+      #
+      #     interface!
+      #
+      #     sig { returns(String) }
+      #     def foo; end
+      #   end
+      class ScopeHelperPosition < RuboCop::Cop::Cop
+        include RangeHelp
+        HELPERS = %i(interface! abstract! final!).freeze
+
+        def autocorrect(node)
+          lambda do |corrector|
+            end_pos = if processed_source[node.source_range.line].blank? &&
+              processed_source[node.source_range.line - 2].blank?
+              node.source_range.end_pos + 1
+            else
+              node.source_range.end_pos
+            end
+
+            range = range_between(node.source_range.begin_pos, end_pos)
+            corrector.remove(range_by_whole_lines(range, include_final_newline: true))
+
+            first_def = node.parent.each_child_node.find(&:def_type?)
```

Comment:
It might be more robust to ensure only `extend`, `prepend` and `include` are before?

Without this you'll also have to deal with calls to singleton methods, blocks, sub-scopes, ...

Reply:
I think @Morriar means "calls to" those methods will still be above where you will ultimately insert the helper method call, as in:
```ruby
class Foo
  extend T::Helpers

  attribute :foo, String
  attribute :bar, Integer

  validate_uniqueness :foo, if: -> do
    # some condition here
  end

  some_method_call_that_takes_a_block do 
    # some long lines of code here
  end

  interface!

  def foo_bar
  end
end
```
the `interface!` call is still too low in the class definition even though it is above the first method call.

Can we ensure that the helper method call is right below the `extend T::Helpers` line and that that itself is right below `extend T::Sig`?

```ruby
class Foo
  extend T::Sig

  some_class_method_call_here

  sig { void }
  def foo_bar

  extend T::Helpers
 
  sealed!
end
```
should be turned into:  
```ruby
class Foo
  extend T::Sig
  extend T::Helpers
 
  sealed!

  some_class_method_call_here

  sig { void }
  def foo_bar

end
```